src/compute/src/render/context.rs

```diff
@@ -181,6 +182,40 @@ where
     }
 }
 
+impl<S: Scope, T> Context<S, T>
+where
+    T: Timestamp + Lattice + Columnation,
+    S::Timestamp: Lattice + Refines<T> + Columnation,
+{
+    /// Brings the underlying arrangement into a region.
+    /// Creates a new region in the context, provides the context
+    pub fn enter_region<'a>(
+        &self,
+        region: &Child<'a, S, S::Timestamp>,
+        bindings: Option<&std::collections::BTreeSet<Id>>,
+    ) -> Context<Child<'a, S, S::Timestamp>, T> {
+        let bindings = self
+            .bindings
+            .clone()
+            .into_iter()
+            .filter(|(key, _)| bindings.as_ref().map(|b| b.contains(key)).unwrap_or(true))
+            .map(|(key, bundle)| (key, bundle.enter_region(region)))
+            .collect();
```

Comment:
I think you can avoid the clone here:

```suggestion
        let bindings = self
            .bindings
            .iter()
            .filter(|(key, _)| bindings.as_ref().map(|b| b.contains(key)).unwrap_or(true))
            .map(|(key, bundle)| (*key, bundle.enter_region(region)))
            .collect();
```

Reply:
Thank you! I took this change, and did some more clone removal!